editor/src/components/text-editor/text-editor.tsx

```diff
@@ -158,15 +169,69 @@ export const TextEditorWrapper = React.memo((props: TextEditorProps) => {
   return React.createElement(component, passthroughProps, <span {...editorProps} />)
 })
 
-function setSelectionToEnd(element: HTMLSpanElement) {
-  const range = document.createRange()
-  range.selectNodeContents(element)
-  range.collapse(false)
+async function setSelectionToOffset(
+  element: HTMLSpanElement,
+  scale: number,
+  cursorPosition: Coordinates | null,
+) {
+  const sel = window.getSelection()
+  if (sel == null) {
+    return
+  }
+  if (element.childNodes.length != 1) {
+    return
+  }
+  const textNode = element.childNodes[0]
+  if (textNode.nodeType !== element.TEXT_NODE) {
+    return
+  }
 
-  const selection = window.getSelection()
-  if (selection != null) {
-    selection.removeAllRanges()
-    selection.addRange(range)
+  const setRange = (start: number | null) => {
+    sel.removeAllRanges()
+    const range = document.createRange()
+    range.selectNodeContents(textNode)
+    range.collapse(start != null)
+    if (start != null) {
+      range.setStart(textNode, start)
+    }
+    sel.addRange(range)
+    return range
+  }
+
+  const maxLength = setRange(null).endOffset
+
+  if (cursorPosition != null) {
```

Comment:
nit, but this check could be moved out of the function

Reply:
mmh not really because the same function can be used to just set the caret to the end of the sentence as a default (e.g. you press `t` on a selected editable)